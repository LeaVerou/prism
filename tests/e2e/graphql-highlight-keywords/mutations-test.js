const { expect } = require('../../helper/chai');
const prismFactory = require('../../helper/prism-loader');

const Prism = prismFactory.createInstance(['graphql', 'highlight-keywords']);

describe('E2E test - Prism#highlights mutations correctly', () => {
	it('ex. 1', () => {
		expect(subject(`
mutation orderUpdate($input: OrderInput!) {
	orderUpdate(input: $input) {
		order {
			# Order fields
		}
		userErrors {
			field
			message
		}
	}
}
`)).toMatchSnapshot();
	});
});

function subject(text) {
	return Prism.highlight(text, Prism.languages.graphql, 'graphql');
}
