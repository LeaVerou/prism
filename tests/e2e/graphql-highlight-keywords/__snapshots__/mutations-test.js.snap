// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E test - Prism#highlights mutations correctly ex. 1 1`] = `
<span class="token keyword keyword-mutation">
  mutation
</span>
<span class="token definition-mutation function">
  orderUpdate
</span>
<span class="token punctuation">
  (
</span>
<span class="token variable variable-input">
  $input
</span>
<span class="token punctuation">
  :
</span>
<span class="token class-name atom-input">
  OrderInput
</span>
<span class="token operator">
  !
</span>
<span class="token punctuation">
  )
</span>
<span class="token punctuation">
  {
</span>
<span class="token property-query property-mutation">
  orderUpdate
</span>
<span class="token punctuation">
  (
</span>
<span class="token attr-name">
  input
</span>
<span class="token punctuation">
  :
</span>
<span class="token variable variable-input">
  $input
</span>
<span class="token punctuation">
  )
</span>
<span class="token punctuation">
  {
</span>
<span class="token object">
  order
</span>
<span class="token punctuation">
  {
</span>
<span class="token comment">
  # Order fields
</span>
<span class="token punctuation">
  }
</span>
<span class="token object">
  userErrors
</span>
<span class="token punctuation">
  {
</span>
<span class="token property">
  field
</span>
<span class="token property">
  message
</span>
<span class="token punctuation">
  }
</span>
<span class="token punctuation">
  }
</span>
<span class="token punctuation">
  }
</span>
`;
